app/controllers/arts_controller.rb

```diff
@@ -58,4 +64,14 @@ def art_params
       :artist
     )
   end
+
+  def authenticate_user
+    if !user_signed_in?
+      raise_error
+    end
+  end
+
+  def raise_error
+    raise ActionController::RoutingError.new("Not Found")
+  end
```

Comment:
Nice job with these methods, but since they are authenticating a particular user they should be defined on the user model and then called here in the controller as you have at the top of the controller class.